src/parameters.jl

```diff
@@ -149,3 +149,43 @@ function flatten(::Type{T}, x::Deferred) where T<:Real
     unflatten_Deferred(v_new::Vector{T}) = Deferred(x.f, unflatten(v_new))
     return v, unflatten_Deferred
 end
+
+"""
+    nearest_orthogonal_matrix(X::StridedMatrix)
+
+Project `X` onto the closest orthogonal matrix in Frobenius norm.
+
+Originally used in varz: https://github.com/wesselb/varz/blob/master/varz/vars.py#L446
+"""
+@inline function nearest_orthogonal_matrix(X::StridedMatrix)
```

Comment:
Does this work for complex matrices?

Reply:
Seems to:
```julia
julia> Y = ParameterHandling.nearest_orthogonal_matrix(X);

julia> Y
3×2 Array{Complex{Float64},2}:
   0.199909-0.135974im  -0.357316+0.594604im
   0.441943-0.720861im   0.360216-0.131677im
 -0.0333523-0.474849im  -0.554495+0.253402im

julia> Y'Y
2×2 Array{Complex{Float64},2}:
         1.0+0.0im        6.93889e-17-6.245e-17im
 6.93889e-17+6.245e-17im          1.0+0.0im
```
I'll see if I can figure out how to test this.

Reply:
I've restricted `nearest_orthogonal_matrix` to `Union{Real, Complex}` and added tests, but it's not possible at present to generalise `orthogonal` to handle everything because we don't have `flatten` implemented for vectors of complex numbers. We could definitely add this, but it would require more work / thought, so I'm leaving `orthogonal` constrained to reals.